src/muz/spacer/spacer_context.cpp

```diff
@@ -3321,6 +3347,85 @@ void context::predecessor_eh()
     }
 }
 
+// For all predecessor predicates in r, unset tags of reachable facts that are
+// not satisfied by the model
```

Comment:
Suggested:
Update a partial model to be consistent over reachable facts.

Conceptually, a reachable fact is asserted as `tag ==> fact`, where `tag` is 
a Boolean literal and `fact` a formula. When the model is partial, it is possible 
that `tag` is `true`, but `model.eval(fact)` is `unknown`. This function fixes the 
model by flipping the value of `tag` in this case.